From 735dd2d07591cb7c93110663be85e202ed8c1b99 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Wed, 10 Sep 2014 18:28:45 -0400
Subject: [ethernet] ixgbe: move setting rx_pb_size into get_invariants

Message-id: <1410373750-11341-26-git-send-email-jogreene@redhat.com>
Patchwork-id: 93531
O-Subject: [RHEL7.1 PATCH 25/50] ixgbe: move setting rx_pb_size into get_invariants
Bugzilla: 1091123
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

BZ: 1091123
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

Signed-off-by: Jacob Keller <jacob.e.keller@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 6997d4d1e629c23d01c3e66425f716f59e22e92e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_82598.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_82598.c
index 15506f0..650d7af 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_82598.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_82598.c
@@ -104,6 +104,7 @@ static s32 ixgbe_get_invariants_82598(struct ixgbe_hw *hw)
  mac->mcft_size = IXGBE_82598_MC_TBL_SIZE;
  mac->vft_size = IXGBE_82598_VFT_TBL_SIZE;
  mac->num_rar_entries = IXGBE_82598_RAR_ENTRIES;
+ mac->rx_pb_size = IXGBE_82598_RX_PB_SIZE;
  mac->max_rx_queues = IXGBE_82598_MAX_RX_QUEUES;
  mac->max_tx_queues = IXGBE_82598_MAX_TX_QUEUES;
  mac->max_msix_vectors = ixgbe_get_pcie_msix_count_generic(hw);
@@ -205,8 +206,6 @@ static s32 ixgbe_start_hw_82598(struct ixgbe_hw *hw)
   IXGBE_WRITE_REG(hw, IXGBE_DCA_RXCTRL(i), regval);
  }
 
- hw->mac.rx_pb_size = IXGBE_82598_RX_PB_SIZE;
-
  /* set the completion timeout for interface */
  if (ret_val == 0)
   ixgbe_set_pcie_completion_timeout(hw);
diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_82599.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_82599.c
index 4083d8e..a80a85f 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_82599.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_82599.c
@@ -270,6 +270,7 @@ static s32 ixgbe_get_invariants_82599(struct ixgbe_hw *hw)
  mac->mcft_size = IXGBE_82599_MC_TBL_SIZE;
  mac->vft_size = IXGBE_82599_VFT_TBL_SIZE;
  mac->num_rar_entries = IXGBE_82599_RAR_ENTRIES;
+ mac->rx_pb_size = IXGBE_82599_RX_PB_SIZE;
  mac->max_rx_queues = IXGBE_82599_MAX_RX_QUEUES;
  mac->max_tx_queues = IXGBE_82599_MAX_TX_QUEUES;
  mac->max_msix_vectors = ixgbe_get_pcie_msix_count_generic(hw);
@@ -2025,7 +2026,6 @@ static s32 ixgbe_start_hw_82599(struct ixgbe_hw *hw)
 
  /* We need to run link autotry after the driver loads */
  hw->mac.autotry_restart = true;
- hw->mac.rx_pb_size = IXGBE_82599_RX_PB_SIZE;
 
  if (ret_val == 0)
   ret_val = ixgbe_verify_fw_version_82599(hw);
diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c
index c870f37..eed790a 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c
@@ -61,6 +61,7 @@ static s32 ixgbe_get_invariants_X540(struct ixgbe_hw *hw)
  mac->mcft_size = IXGBE_X540_MC_TBL_SIZE;
  mac->vft_size = IXGBE_X540_VFT_TBL_SIZE;
  mac->num_rar_entries = IXGBE_X540_RAR_ENTRIES;
+ mac->rx_pb_size = IXGBE_X540_RX_PB_SIZE;
  mac->max_rx_queues = IXGBE_X540_MAX_RX_QUEUES;
  mac->max_tx_queues = IXGBE_X540_MAX_TX_QUEUES;
  mac->max_msix_vectors = ixgbe_get_pcie_msix_count_generic(hw);
@@ -187,7 +188,6 @@ static s32 ixgbe_start_hw_X540(struct ixgbe_hw *hw)
   goto out;
 
  ret_val = ixgbe_start_hw_gen2(hw);
- hw->mac.rx_pb_size = IXGBE_X540_RX_PB_SIZE;
 out:
  return ret_val;
 }
-- 
1.7.1